miner/worker.go

```diff
@@ -188,11 +203,11 @@ func (w *worker) commitNewWork() (*types.Block, error) {
 	}
 	if len(localTxs) > 0 {
 		txs := types.NewTransactionsByPriceAndNonce(env.signer, localTxs, header.BaseFee)
-		w.commitTransactions(env, txs, w.coinbase)
+		w.commitTransactions(env, txs, header.Coinbase)
 	}
 	if len(remoteTxs) > 0 {
 		txs := types.NewTransactionsByPriceAndNonce(env.signer, remoteTxs, header.BaseFee)
-		w.commitTransactions(env, txs, w.coinbase)
+		w.commitTransactions(env, txs, header.Coinbase)
```

Comment:
Shouldn't this be the same as `w.coinbase` ? Can we avoid changing this line?

Reply:
we don't actuall change `w.coinbase` according to `GetCoinbaseAt`. That's because if node operator defines a fee recipient address we should keep it without modifying. `w.coinbase` is set once in `Initialize`. So if we modify it in this function and then fee recipients are enabled then the node operator's initial custom fee recipients adress won't be available. So coinbase in headers can be changed according to the rules in the conract but the coinbase in the miner should not change so we can re-use a custom fee address if possible.